src/SamplesApp/SamplesApp.UITests/TestFramework/ImageAssert.cs

```diff
@@ -77,13 +77,33 @@ public static void AreAlmostEqual(ScreenshotInfo expected, Rectangle expectedRec
 			PixelTolerance tolerance,
 			int line)
 		{
-			using (var actualBitmap = new Bitmap(actual.File.FullName))
+			using var actualBitmap = new Bitmap(actual.File.FullName);
+			AreEqualImpl(expected, expectedRect, actual, actualBitmap, actualRect, expectedToActualScale, tolerance, line);
+		}
+
+		private static void AreEqualImpl(
+			ScreenshotInfo expected,
+			Rectangle expectedRect,
+			ScreenshotInfo actual,
+			Bitmap actualBitmap,
+			Rectangle actualRect,
+			double expectedToActualScale,
+			PixelTolerance tolerance,
+			int line)
+		{
+			var (areEqual, context) = EqualityCheck(expected, expectedRect, actual, actualBitmap, actualRect, expectedToActualScale, tolerance, line);
+
+			if (areEqual)
+			{
+				Console.WriteLine(context.ToString());
+			}
+			else
 			{
-				AreEqualImpl(expected, expectedRect, actual, actualBitmap, actualRect, expectedToActualScale, tolerance, line);
+				Assert.Fail(context.ToString());
 			}
 		}
 
-		private static void AreEqualImpl(
+		private static (bool areEqual, string context) EqualityCheck(
```

Comment:
You should inject [CallerMemberName] here, otherwise it will always prints "ImageAssert.AreEqual".

Reply:
This is an private method, won't be called from itself, but sure. it will avoid future errors :)